core/templates/dev/head/domain/statistics/FeedbackImprovementCardObjectFactorySpec.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the FeedbackImprovementCardObjectFactory.
+ */
+
+require('domain/statistics/FeedbackImprovementCardObjectFactory.ts');
+
+describe('FeedbackImprovementCardObjectFactory', function() {
+  var $q = null;
+  var $rootScope = null;
+  var FeedbackImprovementCardObjectFactory = null;
+  var ThreadDataService = null;
+  var FEEDBACK_IMPROVEMENT_CARD_TYPE = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.inject(function(
```

Comment:
Make sure to test everything -- getDirectiveData, getActionButtons, etc.

Reply:
Done.

Reply:
Done.